Bluetooth: controller: Fix missing auxiliary advertising start

cvinayak opened the pull request:
Fix missing auxiliary channel advertising start when
starting Periodic Advertising.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Decouple primary and secondary PDU update

Decouple updates to primary and secondary AD Data PDU such
that primary PDU changes can be committed on successful
scheduling of secondary PDU radio events.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Fix incorrect return value data type

Fix incorrect use of unint32_t instead of a sufficient
uint8_t for the returned HCI error code.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Minor refactor to move ull_hdr_init

Minor refactor to move ull_hdr_init into adv_aux_start and
adv_sync_start functions.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Change indentation

Minor change to indentation of how adv and adv_aux interval
is calculated.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Move ticker status wait

Move ticker status wait into adv_aux_start and
adv_sync_start functions.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Fix missing auxiliary advertising start

Fix missing auxiliary channel advertising start when
starting Periodic Advertising.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Use sys_get_le32 to fetch unaligned value

Use sys_get_le32() to fetch unaligned 32-bit value.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -195,66 +197,131 @@ uint8_t ll_adv_sync_enable(uint8_t handle, uint8_t enable)
 
 	return 0;
+
+sync_cleanup:
+	/* Remove sync_info from auxiliary PDU */
+	err = ull_adv_aux_hdr_set_clear(adv, 0,
+					ULL_ADV_PDU_HDR_FIELD_SYNC_INFO,
+					NULL, NULL, &pri_idx);
+	if (err) {
+		if (err_val) {
+			return err_val;
+		}
+
+		return err;
+	}
+
+	lll_adv_data_enqueue(&adv->lll, pri_idx);
+
+	if (sync->is_started) {
+		/* TODO: we removed sync info, but if sync_stop() fails, what do
+		 * we do?
+		 */
+		err = sync_stop(sync);
+		if (err) {
+			if (err_val) {
+				return err_val;
+			}
+
+			return err;
+		}
+
+		sync->is_started = 0U;
+	}
+
+	if (!enable) {
+		sync->is_enabled = 0U;
+	}
+
+	return err_val;
```
Please move this into a separate function. I do not think the potential savings warrant the use of goto in this case.

cvinayak replied:
Moved it to an inline function.

andrzej-kaczmarek reviewed (approved):
lgtm, just some nitpick

cvinayak committed:
Bluetooth: controller: Rework handling of failure to start sync

Rework implementation handling the disable and the error
handling when failure to start Periodic Advertising
instance.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:
```diff
@@ -812,17 +810,17 @@ uint32_t ull_adv_aux_evt_init(struct ll_adv_aux_set *aux)
-			   uint32_t volatile *ret_cb)
+			   uint32_t ticks_slot_overhead)
 {
+	uint32_t volatile ret_cb;
```
agreed. But will keep it for the sake of conveying that the value is updated in ISR or another thread.

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -195,63 +198,93 @@ uint8_t ll_adv_sync_enable(uint8_t handle, uint8_t enable)
+			ret = ull_adv_aux_start(aux, ticks_anchor_aux,
+						ticks_slot_overhead_aux);
+			if (ret) {
+				err = sync_remove(sync, adv, 1U,
+					BT_HCI_ERR_INSUFFICIENT_RESOURCES);
+				return err;
```
```suggestion
				sync_remove(sync, adv, 1U);
				return BT_HCI_ERR_INSUFFICIENT_RESOURCES;
```

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
+					ULL_ADV_PDU_HDR_FIELD_SYNC_INFO,
+					NULL, NULL, &pri_idx);
+	if (err) {
+		if (err_val) {
+			return err_val;
+		}
+
```
```suggestion
```

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
+		 */
+		err = sync_stop(sync);
+		if (err) {
+			if (err_val) {
+				return err_val;
+			}
```
```suggestion
```

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
+		sync->is_enabled = 0U;
+	}
+
+	return err_val;
```
```suggestion
	return 0U;
```

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
 	return 0;
 }
 
+static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
+				  struct ll_adv_set *adv, uint8_t enable,
+				  uint8_t err_val)
```
```suggestion
static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
				  struct ll_adv_set *adv, uint8_t enable)
```

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -42,7 +42,9 @@ static int init_reset(void);
 static inline void sync_release(struct ll_adv_sync_set *sync);
 static inline uint16_t sync_handle_get(struct ll_adv_sync_set *sync);
-static inline uint8_t sync_stop(struct ll_adv_sync_set *sync);
+static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
+				  struct ll_adv_set *adv, uint8_t enable,
+				  uint8_t err_val);
```
```suggestion
static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
				  struct ll_adv_set *adv, uint8_t enable);
```

wopu-ot reviewed (changes requested):
I believe the error value handling in sync_remove could and should be simplified (see suggestions).

wopu-ot reviewed (approved):
LGTM
